tests/end_to_end/test_blackbox_tracing.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+LINEA_CODE = """import lineapy
+"""
+CODE = """import matplotlib.pyplot as plt
+import numpy as np
+
+size = 256, 16
+dpi = 72.0
+figsize = size[0] / float(dpi), size[1] / float(dpi)
+fig = plt.figure(figsize=figsize, dpi=dpi)
+plt.axes([0, 0, 1, 1], frameon=False)
+
+dash_styles = ["miter", "bevel", "round"]
+
+for i in range(3):
+    plt.plot(
+        i * 4 + np.arange(3),
+        [0, 1, 0],
+        color="blue",
+        dashes=[12, 5],
+        linewidth=8,
+        dash_joinstyle=dash_styles[i],
+    )
+
+plt.xlim(0, 12), plt.ylim(-1, 2)
+plt.xticks([]), plt.yticks([])
+plt.savefig("output/dash_joinstyle.png", dpi=dpi)
+
+"""
+
```

Comment:
Either remove this newline or add one in between LINEA_CODE and CODE

Reply:
good catch 😂  fixed in last commit